apis/duck/v1alpha1/addressable_types.go

```diff
@@ -106,9 +106,15 @@ func (a *Addressable) ConvertFrom(ctx context.Context, from apis.Convertible) er
 	switch source := from.(type) {
 	case *v1.Addressable:
 		a.URL = source.URL.DeepCopy()
+		if a.URL != nil {
```

Comment:
Perhaps
```
if source.URL != nil {
 a.URL = source.URL.DeepCopy()
 a.Hostname = a.URL.Host
}
```
With possible `else a.URL=nil`, but I think it is redundant.

Reply:
if source.URL is nil, DeepCopy just makes it nil. How strongly do you care about it?

```
func (in *URL) DeepCopy() *URL {
	if in == nil {
		return nil
	}
```